qiskit/aqua/algorithms/single_sample/amplitude_estimation/iqae.py

```diff
@@ -0,0 +1,387 @@
+
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+The Iterative Quantum Amplitude Estimation Algorithm.
+"""
+
+
+import logging
+import numpy as np
+from statsmodels.stats.proportion import proportion_confint
+
+from qiskit import ClassicalRegister, QuantumRegister, QuantumCircuit
+from qiskit.aqua import AquaError
+
+from .ae_base import (AmplitudeEstimationBase)
+
+logger = logging.getLogger(__name__)
+
+
+class IterativeAmplitudeEstimation(AmplitudeEstimationBase):
+    """
+    The Iterative Quantum Amplitude Estimation Algorithm.
+    """
+
+    CONFIGURATION = {
+        'name': 'IterativeAmplitudeEstimation',
+        'description': 'Iterative Amplitude Estimation Algorithm',
+        'input_schema': {
+            '$schema': 'http://json-schema.org/draft-07/schema#',
+            'id': 'IterativeAmplitudeEstimation_schema',
+            'type': 'object',
+            'properties': {
+                'epsilon': {
+                    'type': 'number',
+                    'default': 0.01,
+                    'minimum': 0.0
+                },
+                'alpha': {
+                    'type': 'number',
+                    'default': 0.05,
+                    'minimum': 0.0
+                },
+            },
+            'additionalProperties': False
+        },
+        'problems': ['uncertainty'],
+        'depends': [
+            {
+                'pluggable_type': 'uncertainty_problem',
+                'default': {
+                    'name': 'EuropeanCallDelta'
+                }
+            },
+        ],
+    }
+
+    def __init__(self, epsilon, alpha, ci_method='beta', min_ratio=2, a_factory=None,
+                 q_factory=None, i_objective=None):
+        """
+        Initializer.
+
+        Args:
+            epsilon (float): target precision for estimation target a
```

Comment:
May be good to include a note about how the execution scales with respect to epsilon.